### PR TITLE
RSDK-914 - Implement session manager and client support

### PR DESCRIPTION
**Pull request by edaniels**
See JIRA for a copy of the scope. This is ready to review but still needs:

- [x] Tests
- [x] Technical documentation (some from scope)
- [x] Depends on https://github.com/viamrobotics/api/pull/121 and https://github.com/viamrobotics/viam-typescript-sdk/pull/5

### CODE REVIEW

**Comment by edaniels** [comment]
FYI @stevebriskin: not required for review but add yourself if you'd like!

**Review by cheukt — APPROVED** [review]
looks good but definitely want to see some tests

**Comment by edaniels** [comment]
Docs written to https://github.com/viamrobotics/rdk/blob/54b8e0dbadb44a54fe86c49f74b42b90722c9c6f/session/doc.go

File: services/baseremotecontrol/builtin/builtin_test.go [hunk 1]
```diff
@@ -146,7 +146,7 @@ func TestBaseRemoteControl(t *testing.T) {
 		},
 		logger)
 	test.That(t, err, test.ShouldBeError,
-		errors.New("dependency \"baseTest\" should an implementation of <nil> but it was a *inject.InputController"))
+		errors.New("dependency \"baseTest\" should be an implementation of base.Base but it was a *inject.InputController"))
```
Comment (edaniels) [inline services/baseremotecontrol/builtin/builtin_test.go:149]: @Kschappacher fyi

File: components/arm/arm.go [hunk 2]
```diff
@@ -121,17 +121,17 @@ var (
 
 // NewUnimplementedInterfaceError is used when there is a failed interface check.
 func NewUnimplementedInterfaceError(actual interface{}) error {
-	return utils.NewUnimplementedInterfaceError((Arm)(nil), actual)
+	return utils.NewUnimplementedInterfaceError((*Arm)(nil), actual)
```
Comment (edaniels) [inline components/arm/arm.go:124]: @cheukt this was not working

File: config/config.go [hunk 3]
```diff
@@ -146,20 +146,90 @@ func (c *Config) CopyOnlyPublicFields() (*Config, error) {
...
+type remoteData struct {
 	Name                    string                       `json:"name"`
 	Address                 string                       `json:"address"`
 	Frame                   *Frame                       `json:"frame,omitempty"`
 	Auth                    RemoteAuth                   `json:"auth"`
 	ManagedBy               string                       `json:"managed_by"`
 	Insecure                bool                         `json:"insecure"`
-	ConnectionCheckInterval time.Duration                `json:"connection_check_interval,omitempty"`
-	ReconnectInterval       time.Duration                `json:"reconnect_interval,omitempty"`
+	ConnectionCheckInterval string                       `json:"connection_check_interval,omitempty"`
```
Comment (edaniels) [inline config/config.go:171]: @cheukt neither was this

**Pull request merged**